src/Core/Managed/Shared/Extensibility/Metrics/MetricExtractorTelemetryProcessorBase.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Globalization;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public abstract class MetricExtractorTelemetryProcessorBase : ITelemetryProcessor, ITelemetryModule, IDisposable
+    {
+        private MetricManager _metricManager = null;
```

Comment:
Adding to my previous comment. The number of processors you create would define the number of managers you got (and number of timers within them. Not a big deal, but that is one of the "cons" of the approach you have coded up...